### DISCUSSION TITLE
Vue.js

### DISCUSSION THREAD

**Discussion by Igorkowalski94** [Igorkowalski94/eslint-plugin-project-structure#6] [General]
All questions and answers related to Vue.js project structure.

**Comment by oleg-gorbulich** [comment]
good day
Can I pass an absolute link to my folder, Structure.json, into the configuration? I have a monorep and the esLint configuration is transferred to all projects via a link from the root project. It’s hard to copy folderStructure.json to every projects, is it possible to somehow add folderStructure.json directly to the setting or link to it "@rootProject/configs/folderStructure.json"?
I tried this, but it doesn't work
` rules: {
 'project-structure/folder-structure': ['error', folderStructure],
 },`

**Reply by Igorkowalski94** [reply]
.eslintrc

````jsonc
{
  "settings": {
      "project-structure/folder-structure-config-path":
        "...somePath/folderStructure.json",
    },
  "rules": {
      "project-structure/folder-structure": "error",
   }
}
````

try 

````jsonc
{
  "settings": {
      "project-structure/folder-structure-config-path":
        "@rootProject/configs/folderStructure.json",
  },
  "rules": {
      "project-structure/folder-structure": "error",
   }
}
````

**Reply by oleg-gorbulich** [reply]
`export function projectStructureLints() {

  return [

    {
      plugins: {
        'project-structure': structurePlugin,
      },

      rules: {
        'project-structure/folder-structure': 'error',
      },

      settings: {
        'project-structure/folder-structure-config-path':
          'D:/Constr/kit/packages/configs/folderStructure.json',

      },
    },
  ];
}`

errors look like this

`D:\Constr\kit\packages\mockit\src\storage.ts
  2:1  error
🔥 Invalid configuration file 'D:\Constr\kit\packages\mockit\D:/Constr/kit/packages/configs/folderStructure.json' 🔥
  project-structure/folder-structure

D:\Constr\kit\packages\mockit\src\storageRuntime.ts
  1:1  error
🔥 Invalid configuration file 'D:\Constr\kit\packages\mockit\D:/Constr/kit/packages/configs/folderStructure.json' 🔥
  project-structure/folder-structure

D:\Constr\kit\packages\mockit\vite.config.ts
  1:1  error
🔥 Invalid configuration file 'D:\Constr\kit\packages\mockit\D:/Constr/kit/packages/configs/folderStructure.json'`

**Reply by Igorkowalski94** [reply]
Generally, you can also have folderStructure.json and .eslintrc in the root of your monorepo and validate the entire monorepo structure from the root.

````
.
├── 📂 projects
├── 📄 folderStructure.json 
├── 📄 .eslintrc 
└── 📄 ... other monorepo files
````
````
{
    "structure": {
        "children": [
            {
                "name": "projects",
                "children": [
                    // ... structure that projects should meet
                ],
            },
            {
                "extension": "*", // All files located in the root of your monorepo, like package.json, .eslintrc, etc.
            },
        ],
    },
}
````

**Reply by oleg-gorbulich** [reply]
I probably phrased it wrong. I have a root config in a separate project from which the esLint configuration is copied to all other projects via a link.

**Reply by Igorkowalski94** [reply]
Then, as a temporary fix, you can change this line locally  https://github.com/Igorkowalski94/eslint-plugin-project-structure/blob/3cc9b9a700f64aedfeb9ac1362edd03eff6c457b/src/rules/folderStructure/helpers/handleProgram.ts#L21

to
```ts
   const configPath = `${
        settings["project-structure/folder-structure-config-path"]
    }`;
```

 In a few days I can add support for absolute paths.

**Reply by Igorkowalski94** [reply]
Improvement added in 2.0.17 (support for absolute/relative paths for configuration file)

**Reply by oleg-gorbulich** [reply]
Thanks a lot
